### PR TITLE
Tpetra: Fixing #10390

### PR DESCRIPTION
**Pull request by csiefer2**
Issue: #10390 
Tests: Fixing it!
Stakeholder feedback:  Communication from @jennloe

### CODE REVIEW

File: packages/tpetra/core/test/KokkosIntegration/ApiTest.cpp [hunk 1]
```diff
@@ -77,12 +77,12 @@ void ApiTest::incr(std::string key) {
 }
 
 void ApiTest::printAll() {
-  fprintf (stderr, "**************** cuda call analysis ****************\n"); 
-  fprintf (stderr, "call                                           times\n");
+  //  fprintf (stderr, "**************** cuda call analysis ****************\n");
```
Comment (tasmith4) [inline packages/tpetra/core/test/KokkosIntegration/ApiTest.cpp:80]: we can just delete these if we don't need them any more, right?

File: packages/tpetra/core/test/KokkosIntegration/KokkosUVM.cpp [hunk 2]
```diff
@@ -135,7 +135,7 @@ void reportTimings(Teuchos::RCP<Teuchos::StackedTimer> timer) {
 }
 
 int main(int argc, char *argv[]) {
-  Kokkos::ScopeGuard kokkos(argc, argv);
+  Kokkos::initialize(argc, argv);
```
Comment (tasmith4) [inline packages/tpetra/core/test/KokkosIntegration/KokkosUVM.cpp:138]: why doesn't Kokkos::ScopeGuard work anymore?

File: packages/tpetra/core/test/KokkosIntegration/KokkosDeepcopy.cpp [hunk 3]
```diff
@@ -51,18 +54,22 @@
...
+  void run(int argc, char *argv[])  {
     log("Tpetra Kokkos DeepCopy Regression test start");
 
     std::string cudaSync("cudaDeviceSynchronize");
     std::string cudaMemcpy("cudaMemcpy");
     std::string cudaMemcpyAsync("cudaMemcpyAsync");
     ApiTest *counter = ApiTest::getInstance();
     //initialize
+#ifdef HAVE_TPETRACORE_MPI
     MPI_Init(&argc, &argv);
```
Comment (tasmith4) [inline packages/tpetra/core/test/KokkosIntegration/KokkosDeepcopy.cpp:71]: don't we have a Tpetra::ScopeGuard that can take care of some of this logic for us?  But also, shouldn't this initialize/finalize stuff get called in main instead?

File: packages/tpetra/core/test/KokkosIntegration/KokkosDeepcopy.cpp [hunk 4]
```diff
@@ -240,17 +248,28 @@ class DeepCopyTester {
 
 private:
   bool isConsistent;
-  void log(const std::string& msg) {
+  void log(const std::string& msg, ApiTest *counter=0) {
```
Comment (tasmith4) [inline packages/tpetra/core/test/KokkosIntegration/KokkosDeepcopy.cpp:251]: I'd strongly prefer to use nullptr as the default

**Review by tasmith4 — CHANGES_REQUESTED** [review]

File: packages/tpetra/core/test/KokkosIntegration/KokkosUVM.cpp (hunk 2 above)
Comment (csiefer2) [reply packages/tpetra/core/test/KokkosIntegration/KokkosUVM.cpp:138]: It probably does, actually.  I've just never used it before and removed it trying to debug something else.

File: packages/tpetra/core/test/KokkosIntegration/KokkosDeepcopy.cpp (hunk 3 above)
Comment (csiefer2) [reply packages/tpetra/core/test/KokkosIntegration/KokkosDeepcopy.cpp:71]: Yeah, all of this initialize / finalize is an artifact of the way this was written.

File: packages/tpetra/core/test/KokkosIntegration/KokkosDeepcopy.cpp (hunk 4 above)
Comment (csiefer2) [reply packages/tpetra/core/test/KokkosIntegration/KokkosDeepcopy.cpp]: Sure, why not.

**Pull request merged**